src/bm_sim/P4Objects.cpp

```diff
@@ -336,7 +336,32 @@ P4Objects::process_single_param(ActionFn* action_fn,
   } else if (type == "parameters_vector") {
     action_fn->parameter_start_vector();
     for (const auto &cfg_parameter_value : cfg_parameter["value"]) {
-      process_single_param(action_fn, cfg_parameter_value, primitive_name);
+      if (cfg_parameter_value.isArray()) {
+        // In the case of structures with headers, lists, tuples
+        int position = 0;
+        std::vector<std::pair<int,Data>> vec_data{};
+        std::vector<std::pair<int,header_id_t>> vec_header_id{};
+        for (const auto &cfg : cfg_parameter_value) {
+          const auto type_cfg = cfg["type"].asString();
+          if (type_cfg == "hexstr") {
+            vec_data.push_back(std::make_pair(position, Data(cfg["value"].asString())));
+          } else if (type_cfg == "header") {
+            const auto header_name = cfg["value"].asString();
+            auto header_id = get_header_id_cfg(header_name);
+            vec_header_id.push_back(std::make_pair(position, header_id));
+          } else if (type_cfg == "bool") {
+              int val = 0;
+              if (cfg["value"].asString() == "true") {
+                val = 1;
+              }
+              vec_data.push_back(std::make_pair(position, Data(val)));
+          }
+          position++;
+        }
+        action_fn->parameter_push_back_list(std::unique_ptr<List>(new List(vec_data, vec_header_id)));
```

Comment:
I don't think the generic notion of `List` should be introduced at all, at least not in this PR. bmv2 purposely only deals with headers (named structs can be mapped to headers) and is not aware of tuples. We should not change that just for `log_msg`.

Reply:
The main idea of this fix is support for log_msg for headers and structures. Structures which contain header as a field are represented same as lists and tuples in JSON file, so if we want to support them, we will support lists and tuples too.

Reply:
There are 2 issues there IMO:
* it doesn't respect the type-value syntax used in the JSON format (https://github.com/p4lang/behavioral-model/blob/master/docs/JSON_format.md#the-type-value-object), some new `type` should be introduced
* there is some kind of asymmetry as inner-most structs are treated as headers, whereas outer structs are JSON arrays?